File: backend/api/models/workout.py
from datetime import date
from typing import Optional

from pydantic import BaseModel


class WorkoutBase(BaseModel):
    """Base workout model with shared attributes"""

    name: str
    distance: float  # in meters
    moving_time: float  # in seconds
    total_elevation_gain: float  # in meters
    type: str  # e.g., "Run", "Ride"
    start_date: date
    average_pace: float = 0
    average_heartrate: Optional[float] = None
    max_heartrate: Optional[float] = None


class WorkoutCreate(WorkoutBase):
    """Model for creating a new workout"""

    strava_id: str
    user_id: str


class Workout(WorkoutBase):
    """Model for a workout with ID and additional fields"""

    id: int
    strava_id: str
    user_id: str

    model_config = {"from_attributes": True}

File: frontend/src/types/workout.ts
export interface WorkoutData {
  id: string;
  name: string;
  distance: number;       // in meters
  moving_time: number;    // in seconds
  elapsed_time: number;   // in seconds
  total_elevation_gain: number;  // in meters
  type: string;           // e.g., "Run", "Ride"
  start_date: string;     // ISO date string
  average_pace: number;  // in meters/second
  max_speed: number;      // in meters/second
  start_latlng: [number, number] | null;
  end_latlng: [number, number] | null;
  average_heartrate?: number;
  max_heartrate?: number;
}

// You might also need this for the Strava API authentication
export interface StravaAuth {
  access_token: string;
  refresh_token: string;
  expires_at: number;
  athlete: {
    id: number;
    username: string;
    firstname: string;
    lastname: string;
  };
}